core/src/test/java/org/apache/iceberg/TestDataTaskParser.java

```diff
@@ -0,0 +1,147 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TestDataTaskParser {
+  @Test
+  public void testStaticDataTask() {
+    StaticDataTask dataTask = (StaticDataTask) createDataTask();
+    String jsonStr = ScanTaskParser.toJson(dataTask);
+    Assertions.assertThat(jsonStr).isEqualTo(snapshotsDataTaskJson());
+    StaticDataTask deserializedTask = (StaticDataTask) ScanTaskParser.fromJson(jsonStr, true);
+    assertDataTaskEquals(dataTask, deserializedTask);
+  }
+
+  private DataTask createDataTask() {
+    Map<String, String> summary1 =
+        ImmutableMap.of(
+            "added-data-files", "1",
+            "added-records", "1",
+            "added-files-size", "10",
+            "changed-partition-count", "1",
+            "total-records", "1",
+            "total-files-size", "10",
+            "total-data-files", "1",
+            "total-delete-files", "0",
+            "total-position-deletes", "0",
+            "total-equality-deletes", "0");
+
+    Map<String, String> summary2 =
+        ImmutableMap.of(
+            "added-data-files", "1",
+            "added-records", "1",
+            "added-files-size", "10",
+            "changed-partition-count", "1",
+            "total-records", "2",
+            "total-files-size", "20",
+            "total-data-files", "2",
+            "total-delete-files", "0",
+            "total-position-deletes", "0",
+            "total-equality-deletes", "0");
+
+    List<Snapshot> snapshots =
+        Arrays.asList(
+            new BaseSnapshot(
+                1L, 1L, null, 1234567890000L, "append", summary1, 1, "file:/tmp/manifest1.avro"),
+            new BaseSnapshot(
+                2L, 2L, 1L, 9876543210000L, "append", summary2, 1, "file:/tmp/manifest2.avro"));
+
+    return StaticDataTask.of(
+        Files.localInput("file:/tmp/metadata2.json"),
+        SnapshotsTable.SNAPSHOT_SCHEMA,
+        SnapshotsTable.SNAPSHOT_SCHEMA,
+        snapshots,
+        SnapshotsTable::snapshotToRow);
+  }
+
+  private String snapshotsDataTaskJson() {
+    return "{\"task-type\":\"data-task\","
```

Comment:
it would be good to have this pretty-formatted so that it's easier to read

Reply:
scan task parser doesn't support pretty format. hence can't have the expected string to be pretty formatted

Reply:
I expect that we will have several long JSONs like this. Would it be better to have them externalized from the test code to separate txt files?

In one way this is better, as we don't need to find the files when we read the tests, OTOH we will be drowned in static strings soon.